src/worker/storage.ts

```diff
@@ -0,0 +1,48 @@
+import Dexie, { DexieOptions } from "dexie";
+
+const DB_NAME = "dictionary_sources";
+const VERSION = 1;
+
+/**
+ * files Table Scehma
+ * +------------------+
+ * | id (primary key) |
+ * +------------------+
+ * | name             |
+ * +------------------+
+ * | file             |
+ * +------------------+
+ */
+const FILES_TABLE_NAME = "files";
+const FILES_TABLE_SCHEMA = "++id, name, file";
+
+interface StoredFile {
+  id?: number;
+  name: string;
+  contents: ArrayBuffer;
+}
+
+export class PredictiveTextStudioDexie extends Dexie {
+  files: Dexie.Table<StoredFile, number>;
+
+  constructor(options?: DexieOptions) {
+    super(DB_NAME, options);
```

Comment:
Did Dexie put this options argument here intentionally for dependency injection?

Reply:
Yeah! Dexie supports dependency injection, making it easier to test — no need to juggle globals!

And `fake-indexeddb` does NOT store things to disk, that's true, but things are stored in process, which allows you to do this kind of assertion where you store something, and check to see if it's still there, as in:

https://github.com/eddieantonio/predictive-text-studio/blob/1dd06e8a0618027555812267a8f4bc7bb912d555/test/test-storage.ts#L13-L18